src/app/features/card/card.component.ts

```
@@ -0,0 +1,41 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { ChuckJoke } from 'src/app/core/models/chuck-joke.model';
+import { ChuckJokesService } from 'src/app/core/services/chuck-jokes.service';
+
+@Component({
+  selector: 'app-card',
+  templateUrl: './card.component.html',
+  styleUrls: ['./card.component.scss'],
+})
+export class CardComponent implements OnInit {
+  public customName = '';
+  public chuckJoke = {} as ChuckJoke;
+  public error = {} as HttpErrorResponse;
+
+  constructor(private chuckJokesService: ChuckJokesService) {}
+
+  ngOnInit(): void {
+    this.onFetchJokes();
+  }
+
+  private onFetchJokes() {
+    this.chuckJokesService.getRandomJoke(this.customName).subscribe({
+      next: (res) => (this.chuckJoke = res),
+      error: (e) => (this.error = e),
+    });
+  }
+
+  public setCustomName(name: string) {
+    this.customName = name;
+    this.onFetchJokes();
+  }
+
+  public showSnackbar(): boolean {
+    return JSON.stringify(this.error) !== '{}'
```

this name indicates that by calling this method you would show snackbar but in fact it returns a boolean, also you should not directly call methods from html template - it can be a performance hit because it calls this method on every change detection(so basically very often with default change detection, you could add console.log above return and see how many logs there'll be when you move your mouse). The best way is to create a class filed for example `isError` and you set it to true/false manually (in this case - when you fetch joke or if start typing to hide error message)